shell/platform/common/cpp/text_input_model.cc

```diff
@@ -26,10 +26,16 @@ static constexpr char kTextInputAction[] = "inputAction";
 static constexpr char kTextInputType[] = "inputType";
 static constexpr char kTextInputTypeName[] = "name";
 
+#if defined(_MSC_VER)
+// TODO(naifu): This temporary code is to solve link error.(VS2015/2017)
+// https://social.msdn.microsoft.com/Forums/vstudio/en-US/8f40dcd8-c67f-4eba-9134-a19b9178e481/vs-2015-rc-linker-stdcodecvt-error
+std::locale::id std::codecvt<char16_t, char, _Mbstatet>::id;
```

Comment:
The recommended solution there is to use a different type in the conversions; is there a specific reason you're doing this instead?

Reply:
I don't think it's necessary to change the correct code for compatibility with compilers (eg, modify the conversion type). This logic is added to ensure that the correct code is properly compiled and linked.
Microsoft has confirmed that it will solve this problem in a future release, and by that time we only need to remove this logic without having to modify all the code.

Reply:
This is a MSVC issue, not a C++ or clang issue. I don't think it should be modified for MSVC.